config/rbac/metrics_auth_role.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    name: zero-trust-workload-identity-manager
+    app.kubernetes.io/name: zero-trust-workload-identity-manager
+    app.kubernetes.io/created-by: zero-trust-workload-identity-manager
+    app.kubernetes.io/part-of: zero-trust-workload-identity-manager
+    app.kubernetes.io/managed-by: kustomize
+    control-plane: controller-manager
+  name: metrics-auth-role
+rules:
+- apiGroups:
+  - authentication.k8s.io
+  resources:
+  - tokenreviews
+  verbs:
+  - create
```

Comment:
`tokenreviews` create permission also exists in `manager-role`: https://github.com/openshift/zero-trust-workload-identity-manager/blob/65d8f1fae629b58eaf98858edd1851aa4d380c0f/config/rbac/role.yaml#L61-L66

But if it's for separation of concerns we could keep it as-is.